### PR TITLE
[UI refactor] Remove .bind() part2 - Remove .bind in counterfactual folder and part of .bind() in other folders

### PR DESCRIPTION
**Pull request by tongyu-microsoft**
This PR removes .bind() in libs\counterfactuals folder. And also removes part of .bind() in core-ui, counterfactual, dataset-explorer, error-analysis folders.

## Description
The UI refactor work is to remove all .bind() functions. And this PR is part 2. 
part 1 PR link: https://github.com/microsoft/responsible-ai-toolbox/pull/1523.
There will be following PRs to remove other .bind() functions. 
The reason that we want to remove .bind() is that, a different callback is created each time the component renders. So we want to remove .bind() and set lint rule in the end to ban adding .bind() in the future.

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

**Review by xuke444 — APPROVED** [review]

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/Matrix/MatrixArea/MatrixArea.tsx [hunk 1]
```diff
@@ -163,25 +161,17 @@ export class MatrixArea extends React.PureComponent<
...
-          />
-          <MatrixOptions
-            quantileBinning={this.state.quantileBinning}
-            binningThreshold={this.state.numBins}
-            updateQuantileBinning={this.updateQuantileBinning.bind(this)}
-            updateNumBins={this.updateNumBins.bind(this)}
-            isEnabled={this.props.isEnabled}
-          />
-        </Stack>
+        <MatrixAreaOptions
```
Comment (imatiach-msft) [inline libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/Matrix/MatrixArea/MatrixArea.tsx:164]: nice refactor!

File: libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortName.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+export class CohortName extends React.Component<ICohortNameProps> {
+  public render(): React.ReactNode {
+    return (
+      <Stack>
+        <Text>{this.props.cohort.name}</Text>
+        <Stack horizontal tokens={{ childrenGap: "s1" }}>
+          {this.props.index && (
+            <Link onClick={this.onClickEditCohort}>
+              {localization.Interpret.CohortBanner.editCohort}
```
Comment (imatiach-msft) [inline libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortName.tsx:25]: This localization path is weird.  It should be in core-ui.  I guess this goes with issue https://github.com/microsoft/responsible-ai-toolbox/issues/1340

**Review by imatiach-msft — APPROVED** [review]

File: libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortName.tsx (hunk 2 above)
Comment (tongyu-microsoft) [reply libs/core-ui/src/lib/Cohort/ManualCohortManagement/CohortName.tsx:25]: Yeah good point, we can update the localization path when we are working on issue https://github.com/microsoft/responsible-ai-toolbox/issues/1340

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/Matrix/MatrixArea/MatrixArea.tsx (hunk 1 above)
Comment (tongyu-microsoft) [reply libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/Matrix/MatrixArea/MatrixArea.tsx:164]: Thank you!!

**Pull request merged**